kyo-core/shared/src/main/scala/kyo/layers.scala

```diff
@@ -0,0 +1,144 @@
+package kyo
+
+object layers {
+
+  trait Layer[Sin, Sout] { self =>
+    def run[T, S](effect: T > (S with Sout))(implicit fl: Flat[T]): T > (S with Sin)
+
+    final def ++[Sin1, Sout1](other: Layer[Sin1, Sout1]): Layer[Sin with Sin1, Sout with Sout1] =
```

Comment:
I'd like to avoid any symbolic operators in Kyo. Maybe we could use `add` for `++` and `andThen` for `>>>`?

Reply:
Changed `++` to `add` and `>>>` to `chain`.  I'm avoiding `andThen` for now, since that could easily mean what is currently `add` (since `add` just handles applies the first layer *and then* the second).

Whether to keep `add` as `add` and not change it to `andThen` will depend on whether it is effectively commutative so long as it does not change the type of `T` when it handles an effect.